core/src/main/java/org/arquillian/smart/testing/configuration/Configuration.java

```diff
@@ -0,0 +1,279 @@
+    private Scm scm;
+
+    public String[] getStrategies() {
+        return strategies;
+    }
+
+    public void setStrategies(String... strategies) {
+        this.strategies = strategies;
+    }
+
+    public RunMode getMode() {
+        return mode;
+    }
+
+    public void setMode(RunMode mode) {
+        this.mode = mode;
+    }
+
+    public String getApplyTo() {
+        return applyTo;
+    }
+
+    public void setApplyTo(String applyTo) {
+        this.applyTo = applyTo;
+    }
+
+    public boolean isDisable() {
+        return disable;
+    }
+
+    public void setDisable(boolean disable) {
+        this.disable = disable;
+    }
+
+    public boolean isDebug() {
+        return debug;
+    }
+
+    public void setDebug(boolean debug) {
+        this.debug = debug;
+    }
+
+    public Report getReport() {
+        return report;
+    }
+
+    public void setReport(Report report) {
+        this.report = report;
+    }
+
+    public Scm getScm() {
+        return scm;
+    }
+
+    public void setScm(Scm scm) {
+        this.scm = scm;
+    }
+
+    public static Configuration load() {
+        final File parent = Paths.get("").toAbsolutePath().toFile();
+        final File[] files = parent.listFiles((dir, name) -> name.equals(SMART_TESTING_YML) || name.equals(SMART_TESTING_YAML));
+
+        Map<String, Object> yamlConfiguration = new LinkedHashMap<>();
+
+        if (files != null) {
+            if (files.length == 0) {
+                logger.info(
+                    "Config file `" + SMART_TESTING_YAML + "` OR `" + SMART_TESTING_YAML + "` is not found. Using system properties to load "
+                        + "configuration for smart testing.");
+            } else {
+                try (InputStream io = Files.newInputStream(getConfigurationFilePath(files))) {
+                    final Yaml yaml = new Yaml();
+                    yamlConfiguration = (Map<String, Object>) yaml.load(io);
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            }
+        }
+
+        return parseConfiguration(yamlConfiguration);
+    }
+
+    public static Configuration loadPrecalculated() {
+        final File parent = Paths.get("").toAbsolutePath().toFile();
+        final File configFile = new LocalStorage(parent).duringExecution().temporary().file(SMART_TESTING_YML).getFile();
+
+        return loadConfigurationFromFile(configFile);
+    }
+
+    static Configuration loadConfigurationFromFile(File configFile) {
+        try(FileReader fileReader = new FileReader(configFile)) {
+            final Yaml yaml = new Yaml();
+            return yaml.loadAs(fileReader, Configuration.class);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to load configuration from file " + configFile.getPath(), e);
+        }
+    }
+
+    public Path dump() {
+        final LocalStorageFileAction configFile = new LocalStorage(Paths.get("").toFile())
```

Comment:
setting path to the root dir should be done using a parameter in `dump()`;

Reply:
The same can be applied also to the other methods

Reply:
mainly the `loadPrecalculated()` one - otherwise it won't work in the surefire forked mode